src/mlpack/methods/nmf/nmf_main.cpp

```diff
@@ -121,9 +127,15 @@ static void mlpackMain()
     if (CLI::HasParam("initial_w"))
     {
       // Initialization with given W, H matrices.
+#ifdef BINDING_MATRIX_TRANSPOSED
+      GivenInitialization ginit = GivenInitialization(
```

Comment:
Do you think it's worth to restructure the code to avoid the duplication of:

```c++
#ifdef BINDING_MATRIX_TRANSPOSED
      GivenInitialization ginit = GivenInitialization(
          std::move(CLI::GetParam<arma::mat>("initial_h")),
          std::move(CLI::GetParam<arma::mat>("initial_w")));
#else
      GivenInitialization ginit = GivenInitialization(
          std::move(CLI::GetParam<arma::mat>("initial_w")),
          std::move(CLI::GetParam<arma::mat>("initial_h")));
#endif
```

e.g. by introducing a RunAMF method?

Reply:
Looks like a good improvement to me.